### PR TITLE
Make the code compatible with PyTorch 0.4.1

### PR DESCRIPTION
**Pull request by billy-inn**
The code is tested on Penn Treebank. There is a small gap between my results with the results reported in the paper.

**MoS w/o finetune**:
Valid 58.34 Test 56.18
**MoS**: 
Valid 56.83 Test 54.64
**MoS + dynamic evaluation**: 
Valid 49.03 Test: 48.43

### CODE REVIEW

File: dynamiceval.py [hunk 1]
```diff
@@ -95,7 +95,8 @@ def gradstat():
 
     while i < train_data.size(0) - 1 - 1:
         seq_len = args.bptt
-        model.eval()
+        model.train()
+        model.use_dropout = False
```
Comment (sh0416) [inline dynamiceval.py:99]: Why do we turn off dropout?
